### PR TITLE
Fixes #29157 - Fix Style/NegatedIf cop

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: extra/changelog [hunk 1]
```diff
@@ -13,7 +13,7 @@ cmd.each_line do |l|
   date, _, _ = date.split(" ")
 
   id = "#{date}\t#{author}"
-  if not list[id]
+  if !(list[id])
```
Comment (ekohl) [inline extra/changelog]: Can this be `unless list[id]` as well?

File: modules/dhcp_native_ms/dhcp_native_ms_main.rb [hunk 2]
```diff
@@ -95,11 +95,11 @@ def unused_ip(subnet_address, mac_address, from_address, to_address)
         foreman_end_addr_ip = IPAddr.new(to_address)
         dhcp_start_addr_ip = IPAddr.new(dhcp_start_addr)
         dhcp_end_addr_ip = IPAddr.new(dhcp_end_addr)
-        if not (dhcp_start_addr_ip..dhcp_end_addr_ip).cover? foreman_start_addr_ip
+        if !(dhcp_start_addr_ip..dhcp_end_addr_ip).cover? foreman_start_addr_ip
```
Comment (ekohl) [inline modules/dhcp_native_ms/dhcp_native_ms_main.rb]: Logically this is equivalent to `unless` as well, right? Same be low

File: extra/changelog (hunk 1 above)
Comment (mmoll) [reply extra/changelog]: Style/NegatedIf does fix this

File: modules/dhcp_native_ms/dhcp_native_ms_main.rb (hunk 2 above)
Comment (mmoll) [reply modules/dhcp_native_ms/dhcp_native_ms_main.rb]: Style/NegatedIf does fix this

File: extra/changelog [hunk 3]
```diff
@@ -13,7 +13,7 @@ cmd.each_line do |l|
   date, _, _ = date.split(" ")
 
   id = "#{date}\t#{author}"
-  if not list[id]
+  unless (list[id])
```
Comment (ekohl) [inline extra/changelog]: Maybe remove the braces as well?

**Review by ekohl — APPROVED** [review]

File: extra/changelog (hunk 3 above)
Comment (mmoll) [reply extra/changelog]: removed.

**Pull request merged**